File: Java/UberVan.java
import java.util.ArrayList;
import java.util.Map;

public class UberVan extends Car{
    Map<String, Map<String,Integer>> typeCarAccepted;
    ArrayList<String> seatsMaterial;

    public UberVan(String license, Account driver, Map<String, Map<String,Integer>> typeCarAccepted, ArrayList<String> seatsMaterial) {
        super(license,driver);
        this.typeCarAccepted = typeCarAccepted;
        this.seatsMaterial = seatsMaterial;
    }

    @Override
    public void setPassenger(int passenger) {
        if (passenger==6) {
            this.passenger = passenger;
        } else {
            System.out.println("Invalid number!, please set 6 passengers!");
        }
    }

    @Override
    public String getDataCar() {
        if (typeCarAccepted!=null && seatsMaterial!=null) {
            return super.getDataCar()+", TypeCarAccepted: "+typeCarAccepted+", Seat Material: "+seatsMaterial;
        }
        return "Error!";
    }
}

File: JS/UberVan.js
var {Car} = require("./Car");

function UberVan(license, driver, typeCarAccepted, seatsMaterial) {
    this.typeCarAccepted = typeCarAccepted;
    this.seatsMaterial = seatsMaterial;
    Car.call(this,license,driver);
}

UberVan.prototype = new Car();

UberVan.prototype.setPassenger = function(passenger) {
    if (passenger==6)
        this.passenger = passenger;
    else
        console.log("Invalid number!, Please set 6 passengers!");
}

UberVan.prototype.getDataCar = function() {
    if (Car.prototype.getDataCar.call(this)!="Error!" && this.typeCarAccepted!=undefined && this.seatsMaterial!=undefined) {
        return Car.prototype.getDataCar.call(this) + ", Type-Car-Accepted: " + this.typeCarAccepted + ", Seat-Material: "+ this.seatsMaterial;
    }
    return "Error!";
}

module.exports = {
    UberVan: UberVan
}